### PR TITLE
Fix Gemspec, Travis build, support Rails 4.1

### PR DESCRIPTION
**Pull request by carlthuringer**
Got the gem working with my Rails 4.1 setup.

It didn’t seem that the foreign_key constraint was necessary with the AR:Relation directive specifying dependent:destroy.

Also, the Gemspec was missing a list of files to include in the gem, so this gem is useless when downloaded from rubygems.

In order to get travis happy, I needed to add a .travis.yml with some postgres setup, and a rakefile for the default rake test.

But happily, with all the tests running we can reasonably say that this is a good PR. :)

### CODE REVIEW

File: lib/generators/flipper/templates/migration.rb [hunk 1]
```diff
@@ -12,12 +12,10 @@ def self.up
       t.string :value
       t.timestamps null: false
     end
-    add_foreign_key :flipper_gates, :flipper_features, on_delete: :cascade
```
Comment (jesseproudman) [inline lib/generators/flipper/templates/migration.rb:15]: I _believe_ these were required for Rails 3.2 to function properly... Will go look at that in a bit.

Comment (jesseproudman) [inline lib/generators/flipper/templates/migration.rb:15]: Reviewed the code and it doesn't look like I ever did need to touch these - they were working out of the box. What were they breaking for you?

**Comment by jesseproudman** [comment]
Thanks for the work here. One open question in the code and then I'm :+1:

File: lib/generators/flipper/templates/migration.rb (hunk 1 above)
Comment (carlthuringer) [inline lib/generators/flipper/templates/migration.rb:15]: Add_foreign_key wasn't available in rails 4.1, I think the foreigner gem wasn't loading properly. 

What is the FK constraint accomplishing here that dependent destroy doesn't under normal operations? The tests pass without it. 

Cheers,
Carl Thuringer

Comment (jesseproudman) [inline lib/generators/flipper/templates/migration.rb:15]: I'm hesitant to diverge this fork from the main installation here.

Foreigner is what adds that command to the 3.2 installation I was using. How was it failing to load for you?

Maybe these contributions make most sense back on the main https://github.com/bgentry/flipper-activerecord repo which I believe has support for Rails 4.0 today? This particular repo was setup to ensure compatibility with Rails 3.2 (hence the name).

Comment (carlthuringer) [inline lib/generators/flipper/templates/migration.rb:15]: Actually, it [Requires ActiveRecord 4.2+](https://github.com/bgentry/flipper-activerecord/blob/master/flipper-activerecord.gemspec#L19) which adds support for the foreign key function in migrations.

There's no real explanation in the main branch for the foreign key constraint, so I can only determine what I can figure out by reading the guide and google. Apparently this guarantees referential integrity at the database level, which is nice if you delete one record and then suddenly crash and burn... though I wonder if one could call it a critical application of the tech.

```
-- add_foreign_key(:flipper_gates, :flipper_features, {:on_delete=>:cascade})
rake aborted!
An error has occurred, all later migrations canceled:

undefined method `add_foreign_key' for #<CreateFlipperTables:0x007fbfa2632068>/Users/carl/Workspace/gazebo/db/migrate/20150827000518_create_flipper_tables.rb:15:in `up'
```

That's the problem I had.

I tried a few more things after backing off some of my changes here.

What worked was adding `gem "foreigner"` to my Gemfile in the host application before running the migration...

I wonder if there's some way to achieve the same effect without requiring the host application to add the Gem as a direct dependency. Need to read the Bundler docs.

File: flipper-activerecord3dot2.gemspec [hunk 2]
```diff
@@ -4,11 +4,14 @@ Gem::Specification.new do |gem|
   gem.authors       = ["Jesse Proudman", "Blake Gentry"]
   gem.email         = ["jproudman@bluebox.net", "blakesgentry@gmail.com"]
   gem.description   = %q{ActiveRecord 3.2 adapter for Flipper}
-  gem.summary       = %q{ActiveRecord 3.2 adapter for Flipper}
+  gem.summary       = <<-SUMMARY
+    A fork of `flipper-activerecord` which supports ActiveRecord 3.2 with minimal changes.
```
Comment (jesseproudman) [inline flipper-activerecord3dot2.gemspec]: Let's modify this to read:

A fork of `flipper-activerecord` which supports ActiveRecord 3.2 and ActiveRecord 4.1 with minimal changes.

**Comment by jesseproudman** [comment]
Thanks for your work here. This is coming along nicely.

**Comment by carlthuringer** [comment]
I cleaned up all my commits and got things working with foreigner. It builds on travis but needs a postgres user for the local tests. I suppose we'd need to set up a separate database config for local usage, but I feel like this is more than enough for one pull request.

**Comment by jesseproudman** [comment]
:+1:

**Pull request merged**